builtin/providers/nsone/provider.go

```diff
@@ -0,0 +1,47 @@
+package nsone
+
+import (
+	"github.com/bobtfish/go-nsone-api"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Provider returns a terraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"apikey": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("NSONE_APIKEY", nil),
+				Description: descriptions["api_key"],
+			},
+		},
+		ResourcesMap: map[string]*schema.Resource{
+			"nsone_zone":          zoneResource(),
+			"nsone_record":        recordResource(),
+			"nsone_datasource":    dataSourceResource(),
+			"nsone_datafeed":      dataFeedResource(),
+			"nsone_monitoringjob": monitoringJobResource(),
+			"nsone_user":          userResource(),
+			"nsone_apikey":        apikeyResource(),
+			"nsone_team":          teamResource(),
+		},
+		ConfigureFunc: nsoneConfigure,
+	}
+}
+
+func nsoneConfigure(d *schema.ResourceData) (interface{}, error) {
+	n := nsone.New(d.Get("apikey").(string))
+	n.Debug()
+	n.RateLimitStrategySleep()
```

Comment:
Agree, though I think adding that would be an additional functionality, which I would prefer to keep out of this PR.